setup.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash -ex
+# The -e option would make our script exit with an error if any command
+# fails while the -x option makes verbosely it output what it does
+
+# Install Pipenv, the -n option makes sudo fail instead of asking for a
+# password if we don't have sufficient privileges to run it
+sudo -n dnf install -y pipenv
+
+cd /vagrant
+# Install dependencies with Pipenv
+pipenv sync --dev
+
+# run our app. Nohup and "&" are used to let the setup script finish
+# while our app stays up. The app logs will be collected in nohup.out
+nohup pipenv run python manage.py runserver 0.0.0.0:8000 &
```

Comment:
See that red symbol below the last line. It tells you, there's no newline at the end of the file. You should generally have a single newline at the end of the file (no more, no less). Most editors also have configurations for adding that when  files are saved, so you don't have to worry about it.

![image](https://user-images.githubusercontent.com/12252819/200766866-6099bce1-387e-4ff8-b099-ca56f8038489.png)